src/hooks/useQuery.ts

```diff
@@ -0,0 +1,5 @@
+import { useLocation } from "react-router-dom";
+
+export function useQuery() {
+  return new URLSearchParams(useLocation().search);
+}
```

Comment:
it may be a good idea to rename this to `useQueryParams` as this name is very similar to one of the main hooks implemented by [react-query](https://react-query.tanstack.com/guides/queries). This change would be primarily to prevent confusion from contributors